### PR TITLE
feat: add bigint methods

### PR DESCRIPTION
**Pull request by Khasms**

### CODE REVIEW

File: src/validators/BigIntValidator.ts [hunk 1]
```diff
@@ -37,6 +37,22 @@ export class BigIntValidator<T extends bigint> extends BaseValidator<T> {
 		return this.lt(0n);
 	}
 
+	public divisibleBy(number: bigint): this {
+		return this.addConstraint(bigintDivisibleBy(number) as IConstraint<T>);
+	}
+
+	public get abs(): this {
+		return this.transform(((value) => (value < 0 ? -value : value)) as (value: bigint) => T);
```
Comment (kyranet) [inline src/validators/BigIntValidator.ts:45]:
```suggestion
		return this.transform((value) => (value < 0 ? -value : value) as T);
```

File: src/validators/BigIntValidator.ts [hunk 2]
```diff
@@ -37,6 +37,22 @@ export class BigIntValidator<T extends bigint> extends BaseValidator<T> {
...
+	}
+
+	public intN(bits: number): this {
+		return this.transform(((value) => BigInt.asIntN(bits, value)) as (value: bigint) => T);
```
Comment (kyranet) [inline src/validators/BigIntValidator.ts:49]:
```suggestion
		return this.transform((value) => BigInt.asIntN(bits, value) as T);
```

File: src/validators/BigIntValidator.ts [hunk 3]
```diff
@@ -37,6 +37,22 @@ export class BigIntValidator<T extends bigint> extends BaseValidator<T> {
...
+	}
+
+	public uintN(bits: number): this {
+		return this.transform(((value) => BigInt.asUintN(bits, value)) as (value: bigint) => T);
```
Comment (kyranet) [inline src/validators/BigIntValidator.ts:53]:
```suggestion
		return this.transform((value) => BigInt.asUintN(bits, value) as T);
```

File: tests/validators/bigint.test.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
+import { ConstraintError, s, ValidationError } from '../../src';
+
+const safeInteger = 42n;
+// eslint-disable-next-line @typescript-eslint/no-loss-of-precision
+const unsafeInteger = 242043489611808769n;
```
Comment (kyranet) [inline tests/validators/bigint.test.ts:5]: They're always safe, and the eslint ignore is unneeded.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**